src/components/ActivityCollector/createClickedElementProperties.js

```diff
@@ -0,0 +1,213 @@
+    props.linkName = name;
+    props.linkRegion = region;
+    props.linkType = type;
+    props.linkUrl = URL;
+  }
+  // DATA has priority over XDM
+  /* eslint no-underscore-dangle: 0 */
+  if (data && data.__adobe && data.__adobe.analytics) {
+    const { c } = data.__adobe.analytics;
+    if (c && c.a && c.a.activitymap) {
+      // Set the properties if they exists
+      const { page, link, region, pageIDType } = c.a.activitymap;
+      props.pageName = page || props.pageName;
+      props.linkName = link || props.linkName;
+      props.linkRegion = region || props.linkRegion;
+      if (pageIDType !== undefined) {
+        props.pageIDType = pageIDType;
+      }
+    }
+  }
+};
+
+export default properties => {
+  let props = properties || {};
+  const clickedElementProperties = {
+    get pageName() {
+      return props.pageName;
+    },
+    set pageName(value) {
+      props.pageName = value;
+    },
+    get linkName() {
+      return props.linkName;
+    },
+    set linkName(value) {
+      props.linkName = value;
+    },
+    get linkRegion() {
+      return props.linkRegion;
+    },
+    set linkRegion(value) {
+      props.linkRegion = value;
+    },
+    get linkType() {
+      return props.linkType;
+    },
+    set linkType(value) {
+      props.linkType = value;
+    },
+    get linkUrl() {
+      return props.linkUrl;
+    },
+    set linkUrl(value) {
+      props.linkUrl = value;
+    },
+    get pageIDType() {
+      return props.pageIDType;
+    },
+    set pageIDType(value) {
+      props.pageIDType = value;
+    },
+    get clickedElement() {
+      return props.clickedElement;
+    },
+    set clickedElement(value) {
+      props.clickedElement = value;
+    },
+    get properties() {
+      return {
+        pageName: props.pageName,
+        linkName: props.linkName,
+        linkRegion: props.linkRegion,
+        linkType: props.linkType,
+        linkUrl: props.linkUrl,
+        pageIDType: props.pageIDType
+      };
+    },
+    isValidLink() {
+      return (
+        !!props.linkUrl &&
+        !!props.linkType &&
+        !!props.linkName &&
+        !!props.linkRegion
+      );
+    },
+    isInternalLink() {
+      return this.isValidLink() && props.linkType === "other";
+    },
+    isValidActivityMapData() {
+      return (
+        !!props.pageName &&
+        !!props.linkName &&
+        !!props.linkRegion &&
+        props.pageIDType !== undefined
+      );
+    },
+    get xdm() {
+      if (props.filteredXdm) {
+        return props.filteredXdm;
+      }
+      return buildXdmFromClickedElementProperties(this);
+    },
+    get data() {
+      if (props.filteredData) {
+        return props.filteredData;
+      }
+      return buildDataFromClickedElementProperties(this);
+    },
+    applyPropertyFilter(filter) {
+      if (filter && filter(props) === false) {
+        // logger.info(
+        //   `Clicked element properties were rejected by filter function`
+        // );
```

Comment:
What do you think about including the element (or possibly the props) in that msg as well? Keep in mind this could get called a lot if a user clicks around a lot on elements that doesn't cause navigation or view changes.

Reply:
That's fine. Keep in mind we only log if logging is turned on for WEB SDK.